### ISSUE TITLE
process::extract_without_order must avoid re-invoking processor methods

### ISSUE THREAD

**Issue by seanpianka** [logannc/fuzzywuzzy-rs#6]
See the [original implementation](https://github.com/seatgeek/fuzzywuzzy/blob/master/fuzzywuzzy/process.py#L85) where the processor function is only executed once if it's a known method.

I'm not sure if this is an optimization to prevent duplicate calls, or if multiple invocations could lead to some undesired state? i.e. the processor function is not idempotent.

**Comment by logannc** [comment]
`wratio` -> `wratio` and `wratio_calc` like `ratio_full` and `simple_calc`

same with `partial_ratio_full` for that matter

**Comment by logannc** [comment]
basically the current impl of extract_without_order doesn't double call the processors, but if the *scorer* calls processing, you're hosed. It just so happens the original scorers by default do because we havent split them out as concepts.

**Comment by seanpianka** [comment]
Does #28 split them out as concepts?

**Comment by logannc** [comment]
not quite, no. You have things like `ratio_full` which does processor and delegates to `simple_ratio` which could be considered a scorer, but that structure is not applied to, say, `partial_ratio_full` which would need to split out most of its body into a `best_partial_ratio` or something.
